bpy_addon_build/build_context/hook_definitions.py

```diff
@@ -0,0 +1,160 @@
+from typing import Optional, Union
+from bpy_addon_build.api import BpyError, BpyWarning
+from rich.console import Console
+
+from bpy_addon_build.build_context import BuildContext
+
+
+def build_action_prebuild(ctx: BuildContext, action: str, console: Console) -> None:
+    """
+    Runs an action's pre_build function
+
+    ctx: Build context
+    action: string representing the action name
+    console: Console from Rich
+
+    Returns:
+        None
+    """
+    if ctx.config.build_actions is None and len(ctx.cli.actions):
```

Comment:
Minor style comment that here and below, the nesting is getting pretty deep. Would apply to the other hooks.

I'm a fan of early returns, so you could have basically only ever one level of indentnation.

e.g. it starts with
```
if not if action in ctx.cli.actions:
    return
if action not in ctx.api.action_mods:
    print("Action not in API!")
    return
if not hasattr(ctx.api.action_mods[action], "post_install"):
    return
...